---
title: "R plots code"
output: html_notebook
---

This is review of all the code I have used for data visualization.Most of these plots have been used in publications.

```{r}
library(ggthemr)
random_colors<-c("#4a4a48","#DB784D","#95CC5E","#E84646","#F8BB39","#7A7267","#E1AA93" ,"#168E7F","#555ba4","#85919e")

new <- define_palette(
  swatch = random_colors,
  gradient = c(lower = random_colors[1L], upper = random_colors[2L])
)

ggthemr(new)

```


```{r}
CV_NR<-read.table(file="~/Desktop/ModernR/visualization/panel_four_box.txt",header = TRUE,stringsAsFactors = T)
p <- ggplot(CV_NR, aes(x=Model,fill=Model, y=Acc.TotGen)) + geom_boxplot() 
p <- p + facet_wrap(~Trait,scales="free_y")
p <- p + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), panel.border = element_rect(fill=NA, colour = "black"), axis.line = element_line(colour = "black"), axis.title.x = element_text(size=18, face="bold", vjust=-0.25), axis.title.y = element_text(size=18, face="bold", vjust=0.5),axis.text.y = element_text(size=18,face="bold", color="black"), axis.text.x = element_text(size=16,face="bold", color="black",angle=45,vjust=1.0,hjust=1.0), strip.text.x = element_text(size = 16, face="bold"), strip.text.y = element_text(size = 16, face="bold"))

p <- p + labs(x="Trait",y="Accuracy",title="Cross-validation Genetic Gain and Cycle 1")

p

```
```{r}
load("~/Google Drive/NaCCRI-GWASpop/Dosage_RF_GBSallref.Rdata")##new file more ind with 45k RF overlap from guille/ramu
snps<-(snps+1)
snps[1:5,1:20]

pca<-prcomp(K,scale=F)
scores <- data.frame(colnames(K), pca$x[,1:3])
```




```{r}
PCA<-data.frame(ID=rep(0,(nrow(scores)),color=rep(0,(nrow(scores)))))
PCA[,1]<-colnames(K)
PCA[grep("UG14",colnames(K)),2]<-"green" # color coding
PCA[-grep("UG14",PCA$ID),2]<-"darkblue" # color coding
library(ggthemr)
ggthemr('flat')
library(ggplot2)
pc1 <- qplot(x=PC1, y=PC2, data=scores, colour=factor(PCA[,2])) +
  theme(legend.position="none")
pc2 <- qplot(x=PC1, y=PC3, data=scores, colour=factor(PCA[,2])) +
  theme(legend.position="none")
pc3 <- qplot(x=PC2, y=PC3, data=scores, colour=factor(PCA[,2])) +
  theme(legend.position="none") 
library(gridExtra)
grid.arrange(pc1, pc2, pc3,ncol=2, nrow = 2)
```
Line plot with two references in y axis 
read files
```{r}
plot_all=read.table("~/Google Drive/genomic_prediction_paper/bar_gged.txt", header=T, stringsAsFactor=F)
df2 <- read.table("~/Google Drive/genomic_prediction_paper/Allpred_GGed.txt", header=T)

```
This is a library with various color combinations for plots
```{r}
library(ggthemr)
ggthemr('flat')
darken_swatch(amount=0.2)
```

We convert the ggplot commands into a function
```{r}
plot <- function(df,df2,trait)
{
  full<-df2[df2$Trait==trait & df2$category == "Full","Accuracy"]
  cv<-df2[df2$Trait==trait & df2$category == "CV","Accuracy"]
  limits <- aes(ymax = mean + se, ymin=mean - se) 
  ggplotDM<-ggplot[ ggplot$trait ==trait,]
  p <- ggplot(ggplotDM, aes(colour=category, y=mean, x=size))
  p + geom_line(aes(group=category)) + geom_errorbar(limits, width=0.2)+theme_bw()+geom_hline(yintercept = c(full),color="red")+geom_hline(yintercept=cv, color = "orange")+theme(axis.title.x=element_blank(),axis.text.x  = element_text(angle=45, vjust=1.0,hjust=1.0, size=10),axis.title.y =element_blank() ,axis.text.y  = element_text(angle=0, vjust=0.5, size=16))+ggtitle("Vigor")+ scale_y_continuous(limits=c(-0.1,0.3))+ scale_x_continuous(breaks=c(300,600,900))+
  annotate("text", label = "full", x = 250, y = full, size = 5, colour = "black",vjust=-1) + annotate("text", label = "cv", x = 250, y = cv, size = 5, colour = "black",vjust=-1)
  
}
g1<-plot(plot_all,df2,"VIGOR")
g1
```

Correlation plots
CBSD chromosomes
